spec/requests/api/v1/course_spec.rb

```diff
@@ -123,16 +122,22 @@ def get_course_codes_from_body(body)
       end
 
       it 'includes correct next link in response headers' do
-        create(:course,
-               course_code: "LAST1",
-               age: 10.minutes.ago,
-               provider: provider)
+        Timecop.freeze(10.minutes.ago) do
+          first_course = create(:course,
+                                course_code: 'LAST1',
+                                provider: provider)
+
+          create(:site_status, :published, course: first_course)
+        end
```

Comment:
Re: the diff on this file; it's not clear to me from the commit message why this patch on this file is so large. I'm sure it's a good improvement, but I can't tell from reading it without a long time getting into your head why it was needed or desired for this particular PR.

Generally I'd suggest either:

* explanation in the commit message as to why you needed/wanted to improve the file
* overall improvments to the spec that are not specific to the task or that are improvements to be built on should be pulled out into a separate base commit or maybe even a separate PR

Reply:
These changes were done because the tests break without it. In these tests the courses have no `site_statuses` and so this PR causes those courses not to show, since this is correct behaviour it means those courses must be given `site_statuses`. Since creating these associations updates certain things I've used TimeCop to make those guarantees.  
I've improved the commit message to clarify.